anonlink/serialization.py

```diff
@@ -0,0 +1,222 @@
+import array as _array
+import heapq as _heapq
+import io as _io
+import struct as _struct
+import typing as _typing
+
+from . import typechecking as _typechecking
+
+
+# FILE FORMAT
+# This is subject to change.
```

Comment:
Please document your justifications for using an "array of structures" representation for serialisation when the internal `CandidatePairs` is a "structure of arrays" representation. Half of the work done in this file could be removed if you serialised as SoA too; your reasons for not doing so (ease of merging two serialised files without reading everything into memory) should be made clear.